src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/IOption2.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    internal interface IOption2 : IEquatable<IOption2?>
+#if !CODE_STYLE
+        , IOption
+#endif
+    {
+        OptionDefinition OptionDefinition { get; }
+
+#if CODE_STYLE
+        string Feature { get; }
+        string Name { get; }
+        Type Type { get; }
+        object? DefaultValue { get; }
+        bool IsPerLanguage { get; }
+
+        ImmutableArray<OptionStorageLocation2> StorageLocations { get; }
+#endif
```

Comment:
This file seems sufficiently magic as to need some comment in it *somewhere*.

Reply:
Sure, will do in a follow-up PR to avoid re-running all the tests and risking another merge conflict resolution round :-). Hope that is fine.